pkg/load/load.go

```diff
@@ -36,7 +36,33 @@ const (
 	commandsSuffix = "-commands.sh"
 )
 
-// FilenameToConfig contains configs keyed by the file they were found int
+// ByOrgRepo maps org --> repo --> list of branched and variant configs
+type ByOrgRepo map[string]map[string][]api.ReleaseBuildConfiguration
+
+func FromPathByOrgRepo(path string) (ByOrgRepo, error) {
+	byFilename, err := FromPath(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return PartitionByOrgRepo(byFilename), nil
+}
+
+func PartitionByOrgRepo(byFilename FilenameToConfig) ByOrgRepo {
+	byOrgRepo := map[string]map[string][]api.ReleaseBuildConfiguration{}
+	for _, configuration := range byFilename {
+		if _, exists := byOrgRepo[configuration.Metadata.Org]; !exists {
+			byOrgRepo[configuration.Metadata.Org] = map[string][]api.ReleaseBuildConfiguration{}
+		}
+		if _, exists := byOrgRepo[configuration.Metadata.Org][configuration.Metadata.Repo]; !exists {
+			byOrgRepo[configuration.Metadata.Org][configuration.Metadata.Repo] = []api.ReleaseBuildConfiguration{}
+		}
+		byOrgRepo[configuration.Metadata.Org][configuration.Metadata.Repo] = append(byOrgRepo[configuration.Metadata.Org][configuration.Metadata.Repo], configuration)
```

Comment:
semi-nit: I think this method would be more readable if you extracted the org and repo strings to `org` and `repo` variables and used that in the logic. I mean, it's no rocket science, but it *looks* like rocket science :)